UpcomingMovies/UpcomingMovies/UpcomingMoviesPage.xaml.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System.Collections.ObjectModel;
+using System.Linq;
+using System.Web;
+using UpcomingMovies.Component;
+using UpcomingMovies.Consts;
+using UpcomingMovies.Dependency;
+using UpcomingMovies.Model;
+using UpcomingMovies.Parameter;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace UpcomingMovies
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class UpcomingMoviesPage : ContentPage
+    {
+        MovieComponent _movieComponent;
+        MovieParameter _movieParameter;
+
+        #region Properties
+        public ObservableCollection<MovieModel> MovieListObservableCollection { get; set; }
+        public int TotalPages { get; set; }
+        #endregion
+
+        public UpcomingMoviesPage()
+        {
+            MovieListObservableCollection = new ObservableCollection<MovieModel>();
+            _movieComponent = new MovieComponent();
+            _movieParameter = new MovieParameter();
+            _movieParameter.Page = 1;
+            InitializeComponent();
+            SetConfiguration();
+            SetEvents();
+            SetPageData();
+        }
+
+        #region Page Events
+        async void SetPageData()
+        {
+            Device.BeginInvokeOnMainThread(async () =>
+            {
+                try
+                {
+                    _movieComponent.Resource = string.IsNullOrEmpty(_movieParameter.Query) ? MoviesApiResourcesConsts.UPCOMING_MOVIES : MoviesApiResourcesConsts.SEARCH_MOVIE;
+                    var result = await _movieComponent.GetMovies(_movieParameter);
+                    if (!result.Movies.Any())
+                    {
+                        DependencyService.Get<IToast>().ShortToast("No movies found...");
+                        return;
+                    }
+                    PopulateListView(result);
+                }
+                catch
+                {
+                    DependencyService.Get<IToast>().ShortToast("Heck, this was unexpected...");
+                }
+            });
+
+
+        }
+        void SetEvents()
+        {
+            btnSearch.Clicked += BtnSearch_Clicked;
```

Comment:
You never unsubscribe from these events that may cause unexpected behavior. Probably not in this particular case, but it's a good practice to do that.

Reply:
Or I could've defined them on the xaml since they're, by default, disposed when disapearing.